src/rebar_file_utils.erl

```diff
@@ -72,11 +72,14 @@ consult_config(State, Filename) ->
         [T] -> T;
         [] -> []
     end,
-    SubConfigs = [consult_config(State, Entry ++ ".config") ||
-                  Entry <- Config, is_list(Entry)
-                 ],
-
-    [Config | lists:merge(SubConfigs)].
+    lists:flatmap(
+      fun (SubConfig) when is_list(SubConfig) ->
+        case consult_config(State, SubConfig) of
+          [] -> consult_config(State, SubConfig ++ ".config");
```

Comment:
Is this sequence of calls to check whether the file name ends in a .config? It may be cheaper to call something like `lists:suffix(".config", SubConfig)` instead and do a single disk lookup based on that result. Unless you're doing stuff like looking for "file.config.config" as part of your lookups.